azure-spring-boot/src/main/java/com/microsoft/azure/telemetry/TelemetryProxy.java

```diff
@@ -8,91 +8,43 @@
 import com.microsoft.applicationinsights.TelemetryClient;
 import com.microsoft.azure.spring.support.GetHashMac;
 import com.microsoft.azure.utils.PropertyLoader;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.lang.NonNull;
 import org.springframework.util.StringUtils;
 
-import java.util.HashMap;
-import java.util.Iterator;
 import java.util.Map;
 
+@Slf4j
 public class TelemetryProxy {
-    private static final Logger LOGGER = LoggerFactory.getLogger(TelemetryProxy.class);
-    private static final String PROJECT_INFO = "spring-boot-starter/" + PropertyLoader.getProjectVersion();
-
-    protected TelemetryClient client;
-    private boolean isAllowTelemetry;
-
-    public TelemetryProxy(boolean isAllowTelemetry) {
 
-        this.client = new TelemetryClient();
-        this.isAllowTelemetry = isAllowTelemetry;
-
-    }
-
-    public void trackEvent(final String eventName) {
+    private static final String PROJECT_INFO = "spring-boot-starter/" + PropertyLoader.getProjectVersion();
 
-        trackEvent(eventName, null, false);
-    }
+    private final TelemetryClient telemetryClient;
 
-    public void trackEvent(final String eventName, final Map<String, String> customProperties) {
-        trackEvent(eventName, customProperties, false);
+    public TelemetryProxy(String instrumentationKey) {
+        this.telemetryClient = getTelemetryClient(instrumentationKey);
     }
 
-    public void trackEvent(final String eventName, final Map<String, String> customProperties,
-                           final boolean overrideDefaultProperties) {
-
-        Map<String, String> properties = getDefaultProperties();
-        if (this.isAllowTelemetry) {
-            properties = mergeProperties(getDefaultProperties(), customProperties,
-                    overrideDefaultProperties);
+    private TelemetryClient getTelemetryClient(String instrumentationKey) {
+        if (!StringUtils.hasText(instrumentationKey)) {
+            log.warn("Telemetry client instrumentation key must contain text.");
```

Comment:
Not sure whether we need to write `warn` log here instead of some `debug` or `trace` log, will it be a surprise to the users if the users see this?

Reply:
make sense, will change to debug.